src/app/problems/_forms/java-form.ts

```diff
@@ -0,0 +1,43 @@
+import {FormBuilder, FormControl, FormGroup, Validators} from "@angular/forms";
+import {JavaFormData} from "@app/problems/_models/java/java-form-data";
+
+export class JavaForm {
+    /**
+     * Creates a FormGroup for a Java question.
+     */
+    static createForm(): FormGroup {
+        const builder = new FormBuilder();
+        return builder.group({
+            title: new FormControl(null, [Validators.required]),
+            difficulty: new FormControl(null, [Validators.required]),
+            category: new FormControl(null),
+            course: new FormControl(null),
+            event: new FormControl(null),
+            junit_template: new FormControl(null),
+        });
+    }
+
+    /**
+     * Creates a FormGroup for a Java question with existing data.
+     * @param data - The existing data for the question.
+     */
+    static createFormWithData(data: JavaFormData): FormGroup {
```

Comment:
You need `JavaFormData` here you can explicitly add all the parameters.
```
static createFormWithData(question: Question, event: CourseEvent)
```
and create the form from these data.